src/qibolab/platform.py

```diff
@@ -29,104 +32,74 @@ class PlatformSettings:
-        self.pairs: Dict[QubitPairId, QubitPair] = {}
-        for pair in settings["topology"]:
-            pair = tuple(sorted(pair))
-            self.pairs[pair] = QubitPair(self.qubits[pair[0]], self.qubits[pair[1]])
-
-        # register single qubit native gates to ``Qubit`` objects
-        native_gates = settings["native_gates"]
-        for q, gates in native_gates["single_qubit"].items():
-            self.qubits[q].native_gates = SingleQubitNatives.from_dict(self.qubits[q], gates)
-
-        # register two qubit native gates to ``QubitPair`` objects
-        self.two_qubit_native_types = NativeType(0)
-        if "two_qubit" in native_gates:
-            for pair, gatedict in native_gates["two_qubit"].items():
-                pair = tuple(sorted(int(q) if q.isdigit() else q for q in pair.split("-")))
-                self.pairs[pair].native_gates = TwoQubitNatives.from_dict(self.qubits, gatedict)
-                self.two_qubit_native_types |= self.pairs[pair].native_gates.types
-        else:
+    is_connected: bool = False
+    """Flag for whether we are connected to the physical instruments."""
+    two_qubit_native_types: NativeType = field(default_factory=lambda: NativeType(0))
+    """Types of two qubit native gates. Used by the transpiler."""
+    topology: nx.Graph = field(default_factory=nx.Graph)
+    """Graph representing the qubit connectivity in the quantum chip."""
+
+    def __post_init__(self):
+        log.info("Loading platform %s", self.name)
+        if self.resonator_type is None:
+            self.resonator_type = "3D" if self.nqubits == 1 else "2D"
+
+        for pair in self.pairs.values():
+            self.two_qubit_native_types |= pair.native_gates.types
+        if self.two_qubit_native_types is NativeType(0):
             # dummy value to avoid transpiler failure for single qubit devices
             self.two_qubit_native_types = NativeType.CZ
 
-        self.topology: nx.Graph = nx.Graph()
         self.topology.add_nodes_from(self.qubits.keys())
         self.topology.add_edges_from([(pair.qubit1.name, pair.qubit2.name) for pair in self.pairs.values()])
 
-    def __repr__(self):
+    def __str__(self):
         return self.name
 
-    def _check_connected(self):
-        if not self.is_connected:  # pragma: no cover
-            raise_error(RuntimeError, "Cannot access instrument because it is not connected.")
+    @property
+    def nqubits(self) -> int:
+        """Total number of usable qubits in the QPU.."""
+        # TODO: Seperate couplers from qubits (PR #508)
+        return len([qubit for qubit in self.qubits if not (isinstance(qubit, str) and "c" in qubit)])
 
     def dump(self, path: Path):
+        """Serializes the platform and saves it as a yaml file.
+
+        This follows the format explained in :ref:`Using runcards <using_runcards>`.
+
+        Args:
+            path (pathlib.Path): Path that the yaml file will be saved.
+        """
+        from qibolab.utils import dump_qubits
```

Comment:
Why this import is not top-level?

Reply:
It gives a circular import error, I guess because the `Settings` and some types are imported in `qibolab.utils` from `qibolab.platform`.

Conceptually speaking, I think that `platform.dump` should be moved to `qibolab.utils` and be used as
```py
from qibolab.utils import dump

dump(platform, path)
```
because it is the only place left where the `Platform` accepts a path and also the last place in the `Platform` where the TII runcard format is assumed (the rest of the platform is now agnostic of runcards, it only operates with `Qubit`, `QubitPair` and `Instrument` objects). 

`dump` is essentially the inverse of `load_settings + load_qubits` so it makes more sense to be in `utils`. I did not move it because I did not want to break the interface (I think it is used in qibocal), but if you agree maybe we should move it now that we are doing other breaking changes anyway.